### PR TITLE
Always allow invitation by link

### PR DESCRIPTION
**Pull request by yurabakhtin**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix

**The PR fulfills these requirements:**

- [ ] All tests are passing
- [x] Changelog was modified

**Other information:**
https://github.com/humhub/humhub-internal/issues/460

### CODE REVIEW

File: protected/humhub/modules/user/services/LinkRegistrationService.php [hunk 1]
```diff
@@ -57,10 +56,10 @@ public function isValid(): bool
 
     public function isEnabled(): bool
```
Comment (luke-) [inline protected/humhub/modules/user/services/LinkRegistrationService.php:57]: Can't we remove the method then?

**Review by luke- — CHANGES_REQUESTED** [review]
@yurabakhtin Thanks. Please see my remark.

**Review by luke- — CHANGES_REQUESTED** [review]
@yurabakhtin  Would it now still be possible to retrospectively deactivate link invitations, e.g. also via spaces?
e.g. the following scenario.

- Link registration is allowed
- In a space, this is used to an extreme extent. 
- Super Admin wants to deactivate it and ends link registration
---
Would registration with the space registration links still be possible?

**Comment by yurabakhtin** [comment]
@luke- I have tested the Space case.

The feature "Invite by link" is visible on Space only when the general setting "Members can invite external users by link" is enabled:

space-invite

However if I disable the general setting I see the registration is still allowed if user has a link with a correct token.

Admin can reset token to new one by click "Create new link":
reset-token

We could reset ALL tokens(space and general) on each case when the setting "Members can invite external users by link" is updated from "enalbed" to "disabled". But it may be unclear for users because they will see the message:

invalid-token

So it seems we should revert the method `isEnabled()` but use it only for Space case, right?

**Comment by marc-farre** [comment]
@yurabakhtin @luke- I would say:
- no automatic token reset (because we might want to disable the feature for a day and enable it back later, the same links should still be working)
- When "Members can invite external users by link", the full feature is disabled (event previous links are not working anymore)
- Exception: System admins can still invite globally when the feature is disabled

Same behaviour with Email invitations.

But we have an issue with this: the invite link in the People page is the same as the one in the global admin page (for sys-admins).
So it is currently not possible to allow sys-admin links and not links from the People page:

image

image

So I think that ideally, the links should be different...

**Comment by luke-** [comment]
@marc-farre Thanks, that sounds good to me. 

@yurabakhtin Can we implement it like that? Ideally +tests :-)

**Comment by yurabakhtin** [comment]
@marc-farre To be sure what is required I would like to describe how it works now and what I have understood to implement:

Now:

1. We have 2 link token types: Global and Space. The Global is used from Administration and People pages. The Space token is generated per each space.
2. Registration is not blocked from the Global link now even when the setting "Members can invite external users by link" is disabled because Admins always have an access there.

New requirement:
1. Create a new link type: People.
2. Such link should have a different token than Global. I think to add a new param like `target=people` => `/user/registration/by-link?token=ygL9JR8erHfJlq&target=people`
3. Registration should be blocked from such link when "Members can invite external users by link" is disabled.
4. Registration from Global(Admin) link will never be blocked.

**Comment by marc-farre** [comment]
@yurabakhtin Yes, it's exactly that, except that Now:
- Registration IS BLOCKED from the Global link now when the setting "Members can invite external users by link" is disabled despite that the Admins should always have an access there.

**Comment by yurabakhtin** [comment]
> * Registration IS BLOCKED from the Global link now when the setting "Members can invite external users by link" is disabled despite that the Admins should always have an access there.

@marc-farre Sorry, I am confused in this point.
Do you want to block a registration when the setting "Members can invite external users by link" even that Admins have an access to generate such global links?
I thought it was a bug and we should make all global links enabled even when the setting is disabled, because Admins can generate such links and they may don't know the setting is disabled currently, so it will be strange if admin sends such global link to user and he will see error page instead of expected registration form.

Or when you wrote "Now" you mean how it works now without new changes from this PR.
When I wrote "Now" above I meant how it works with new changes from this PR.

**Comment by marc-farre** [comment]
> Or when you wrote "Now" you mean how it works now without new changes from this PR.

I see. I meant currently (before PR). I miss understood "Now". 
Si everything is fine with your summary.

**Comment by yurabakhtin** [comment]
@luke- @marc-farre Done, how it works now in this PR:

- New separate token has been inmplemented for the People page, it is stored as general setting `people.inviteToken`,
- Previous token is stored in the same setting `registration.inviteToken`, it is used for Administration part.
- I didn't add a new param like `target=people` as I wrote above, instead I have decided to detect a target by token value, so the links are different only by token value:
diff_tokens

When the setting is **enabled**:
link-enabled

When **disabled**:
link-disabled

**Comment by marc-farre** [comment]
@yurabakhtin That's really perfect, thanks!

**Review by luke- — APPROVED** [review]

**Comment by luke-** [comment]
@yurabakhtin @marc-farre Thanks, looks fine for me too!

**Pull request merged**